yash-env/src/trap/cond.rs

```diff
@@ -62,14 +63,26 @@ pub struct ParseConditionError;
 /// Conversion from `String` to `Condition`
 ///
 /// This implementation supports parsing uppercase strings like `"EXIT"` and
-/// `"TERM"`.
+/// `"TERM"` as well as signal numbers like `"9"` and `"15"`. The number `"0"`
+/// denotes [`Condition::Exit`].
 impl std::str::FromStr for Condition {
     type Err = ParseConditionError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         // TODO Make case-insensitive
         // TODO Allow SIG-prefix
         // TODO Support real-time signals
+
+        if let Ok(number) = s.parse::<c_int>() {
+            if number == 0 {
+                return Ok(Self::Exit);
+            }
+            return match number.try_into() {
+                Ok(signal) => Ok(Self::Signal(signal)),
+                Err(_) => Err(ParseConditionError),
+            };
+        }
```

Comment:
The expression `number.try_into()` correctly returns an error since the `TryFrom<c_int>` implementation for `Signal` rejects negative numbers, so there is no need to handle negative numbers explicitly.